File: src/main/java/kr/mjc/rachel/spring/jdbc/post/ListPosts.java
package kr.mjc.rachel.spring.jdbc.post;

import kr.mjc.rachel.basics.jdbc.post.Post;
import kr.mjc.rachel.basics.jdbc.post.PostDao;
import kr.mjc.rachel.spring.jdbc.AppConfig;
import lombok.extern.slf4j.Slf4j;
import org.springframework.context.ApplicationContext;
import org.springframework.context.annotation.AnnotationConfigApplicationContext;

import java.util.List;

@Slf4j
public class ListPosts {
  public static void main(String[] args) {
    ApplicationContext applicationContext =
        new AnnotationConfigApplicationContext(AppConfig.class);
    PostDao postDao = applicationContext.getBean(PostDao.class);

    List<Post> postList = postDao.listPosts(5, 1);
    log.info(postList.toString());
  }
}

File: src/main/kotlin/kr/mjc/jacob/spring/jpa/post/ListPosts.kt
package kr.mjc.jacob.spring.jpa.post

import kr.mjc.jacob.spring.jpa.applicationContext
import org.slf4j.LoggerFactory
import org.springframework.data.domain.Page
import org.springframework.data.domain.PageRequest
import org.springframework.data.domain.Sort
import java.util.*

fun main() {
  val postRepository = applicationContext.getBean(PostRepository::class.java)
  val log = LoggerFactory.getLogger({}.javaClass)

  print("List Posts - pageNumber(0~) pageSize ? ")
  val pageable = Scanner(System.`in`).use {
    PageRequest.of(it.nextInt(), it.nextInt(),
        Sort.by(Sort.Direction.DESC, "id"))
  }

  val page: Page<Post> = postRepository.findAll(pageable)
  page.forEach { post ->
    println(
        "id=${post.id}, title=${post.title}, user(id=${post.user.id}, username=${post.user.username})")
  }
  log.info("isFirst={}, isLast={}, totalElements={}, totalPages={}",
      page.isFirst, page.isLast, page.totalElements, page.totalPages)
}